src/python/build_management/build_manager.py

```diff
@@ -487,32 +493,37 @@ def delete(self):
 class Build(BaseBuild):
   """Repesents a build type at a particular revision."""
 
-  def __init__(self, base_build_dir, revision):
+  def __init__(self, base_build_dir, revision, build_prefix='APP'):
     super(Build, self).__init__(base_build_dir)
     self.revision = revision
+    self.build_prefix = build_prefix
 
   def _reset_cwd(self):
     """Reset current working directory. Needed to clean up build
     without hitting dir-in-use exception on Windows."""
     root_directory = environment.get_value('ROOT_DIR')
     os.chdir(root_directory)
 
+  def _delete_partial_build_file(self):
+    """Deletes partial build file (if present). This is needed to make sure we
+    clean up build directory if the previous build was partial."""
+    partial_build_file_path = os.path.join(self.build_dir, PARTIAL_BUILD_FILE)
+    if os.path.exists(partial_build_file_path):
+      self.delete()
+
   def _pre_setup(self):
     """Common pre-setup."""
     self._reset_cwd()
```

Comment:
These two calls, as well as `_setup_build_directories` below aren't needed for an auxiliary build, but seem harmless. I'd rather avoid calling those anyway, but suspect that you might be against adding `is_auxiliary_build` flag to the `Build` class as well. Let me know what you think :)

Reply:
Yes, let's just keep them as is. We should avoid creating special cases.